types/messaging.ts

```diff
@@ -1,9 +1,37 @@
-// TODO: This needs expanding
-interface MessageTrigger {
-  /** Name of the trigger */
-  id: string;
-  /** Additional parameters of the trigger */
-  params?: Array<string|number>
+import { TriggerAction } from "./triggerAction";
+import { MessagingAction } from "./specialMessageAction";
+
+type FluentId = { string_id: string };
+type FluentOrString = FluentId | string;
+
+/** Button label and action */
+interface CFRButton {
+  label: {
+    value: string;
+    attributes: {
+      /** A single character to be used as a shortcut key for the secondary button. This should be one of the characters that appears in the button label */
+      accesskey: string;
+    }
+  } | FluentId,
+  action: MessagingAction;
+}
+
+interface SimpleCFRContent {
+  /** The text that shows up in the chiclet. 20 characters max. */
+  notification_text: FluentOrString;
+  /** The background color of the chiclet as a HEX code. */
+  chiclet_color?: string;
+  heading_text: FluentOrString;
+  text: FluentOrString;
+  /** The icon displayed in the pop-over. Should be 32x32px or 64x64px and png/svg */
+  icon: string;
+  /** Dark theme variant of the icon if needed */
+  icon_dark_theme?: string;
+  /** The label and functionality for the buttons in the pop-over */
+  buttons: {
+    primary: CFRButton;
+    secondary: Array<CFRButton>;
```

Comment:
I can't remember, do we require two button definitions here? If so, this would be a`tuple` not an `Array`, i.e., `[CFRButton, CFRButton]`

Reply:
It's up to us in a way: the code accepts any number of buttons (even for primary) which will then convert into a dropdown list. We only happen to use 2 the secondary action.